org/mozilla/jss/tests/Base64Parsing.java

```diff
@@ -0,0 +1,16 @@
+package org.mozilla.jss.tests;
+
+import org.mozilla.jss.netscape.security.util.Utils;
+
+public class Base64Parsing {
+	public static String mime =     "MIICjDCCAXQCAQIwADCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBALQh\nUMzz+5gLdL9X5XqGwCmjoC5vWFC/g1MV/7YhGhqoIuZfbZav9SGjGm/ty++efk+j\ngkDuB1erJsCkAsxI9GMDH3cbhKsVRsTnznlgy6dhZGFjYSSuwDCYFfFP+kxBjFTL\n+WarXCRc1sKe8ldA8M+IavntwH24IffiEXmjhjiJQlQAFyi0d4v7Jnbs4CZoGONr\n4Wac53U8VEoCDxONZ2DYY0Ds88X+daD/W+qWZpNYbY1LDpPaMwZ9Ll3bHdp6QthE\nfcePk9chpKcQjegjSYXzLNFm1uXUPsADwrg0MGxjMWsWglqOMHTR0ESQE+CAKhcK\neuKhy0RNcUfrDaFfHZECAwEAAaBHMEUGCSqGSIb3DQEJDjE4MDYwNAYDVR0RBC0w\nK4Ipdm0tMTcxLTEyNC5hYmMuaWRtLmxhYi5lbmcuYnJxLnJlZGhhdC5jb20wDQYJ\nKoZIhvcNAQELBQADggEBABZGy1BXwWQfz9KyZPjcSq3bxofQOVQ+k8wRe9etKv/L\nZOGhBDs3NCiTNjGaioChVMQIpieudnUD1oLO12AHlqSjkqEl9cVMbDOcvAipA0h/\n3lPnc10fxPYu/As8Ph3V0cakSm0tckqRqEQyqVdYqk9ouVDND7WyffhwLGwjAsPy\nompzeZ5OwL6rWafVrho+ETkTktz3mvs7VEcyTVR9t1EoYWKXa82Cn/2SbXoZV48T\nn/LqQZvKToKymtUTfXp341F1qdKtJpacFLe2RAxfUsIOSERhF8YeyKA78igDK6UK\n+7xo562+wAhy2w3ryG2qI7s3adycl6+0NP45kWWR80k=\n";
+	public static String url_safe = "MIICjDCCAXQCAQIwADCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBALdtHPsAesaTlWRE3pyW6PtDhbQSUt_jYbKSxLMR7uln7YNiL74pYHFNZ9ZrW51GWPF0Kqldtx6c3zAweynehJtI_AR5pDtHi8bh8L8mqLpsJXgSx5laAGt2n_aAXNxJNS854b_q6lV7k6nIfECsZwTkQV3LCEISjbW9jNLQsuSAV_TznJ3UoEmZh9YMA-cwpBBonLcHxRx-0ymq9R53yUCyAUfLWl0pbfUlCF0EKHJ25et_T8TftzMqdLFJ23gyONdjzgPhc0fNN4fKYqvisg_9V-v0NaVR8VtV1yOhRZp4qy-7JCC6veCb1BZtAVfYJYA4Mj-k9Una6z6Qnjl97VECAwEAAaBHMEUGCSqGSIb3DQEJDjE4MDYwNAYDVR0RBC0wK4Ipdm0tMTcxLTEyNC5hYmMuaWRtLmxhYi5lbmcuYnJxLnJlZGhhdC5jb20wDQYJKoZIhvcNAQELBQADggEBAImj0y3AaGwCxlkgkGtlou4eookqD_xnaeckmS3cil5MA6Ff1Ec0orcYBTNBIGwjQN7Z7FC_7ALOa4E94AfqFl6kquy4ArxrVsP3OIC5kGE6kLcz6F2MJCRuNQdGdxBujlCYyjv4Wr67rLgzeSUD-DGPMehFvdQxTW8VorKBaIxxtFtqiFZtEs_9bSVfw0jzC4a5maM77kl819I9TXNwwEjGd3_eNmab42qvp9KvXS6YhTD-sqE5GmL09YvV3KR1vU1o-9iGJF9Nvg8xOROzTCN-U-TKLv982kbOrrKYwdgnfYlMwsGlnPuwSNhMH51fxZU5o_F6EXOAdLqIEPEuFQo";
```

Comment:
I think these aren't originally the same data. Could we use the data below and compare the decoded data to make sure they match?

URL-safe Base64:
```
MIICjDCCAXQCAQIwADCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBALdt
HPsAesaTlWRE3pyW6PtDhbQSUt_jYbKSxLMR7uln7YNiL74pYHFNZ9ZrW51GWPF0
Kqldtx6c3zAweynehJtI_AR5pDtHi8bh8L8mqLpsJXgSx5laAGt2n_aAXNxJNS85
4b_q6lV7k6nIfECsZwTkQV3LCEISjbW9jNLQsuSAV_TznJ3UoEmZh9YMA-cwpBBo
nLcHxRx-0ymq9R53yUCyAUfLWl0pbfUlCF0EKHJ25et_T8TftzMqdLFJ23gyONdj
zgPhc0fNN4fKYqvisg_9V-v0NaVR8VtV1yOhRZp4qy-7JCC6veCb1BZtAVfYJYA4
Mj-k9Una6z6Qnjl97VECAwEAAaBHMEUGCSqGSIb3DQEJDjE4MDYwNAYDVR0RBC0w
K4Ipdm0tMTcxLTEyNC5hYmMuaWRtLmxhYi5lbmcuYnJxLnJlZGhhdC5jb20wDQYJ
KoZIhvcNAQELBQADggEBAImj0y3AaGwCxlkgkGtlou4eookqD_xnaeckmS3cil5M
A6Ff1Ec0orcYBTNBIGwjQN7Z7FC_7ALOa4E94AfqFl6kquy4ArxrVsP3OIC5kGE6
kLcz6F2MJCRuNQdGdxBujlCYyjv4Wr67rLgzeSUD-DGPMehFvdQxTW8VorKBaIxx
tFtqiFZtEs_9bSVfw0jzC4a5maM77kl819I9TXNwwEjGd3_eNmab42qvp9KvXS6Y
hTD-sqE5GmL09YvV3KR1vU1o-9iGJF9Nvg8xOROzTCN-U-TKLv982kbOrrKYwdgn
fYlMwsGlnPuwSNhMH51fxZU5o_F6EXOAdLqIEPEuFQo
```
MIME Base64:
```
MIICjDCCAXQCAQIwADCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBALdt
HPsAesaTlWRE3pyW6PtDhbQSUt/jYbKSxLMR7uln7YNiL74pYHFNZ9ZrW51GWPF0
Kqldtx6c3zAweynehJtI/AR5pDtHi8bh8L8mqLpsJXgSx5laAGt2n/aAXNxJNS85
4b/q6lV7k6nIfECsZwTkQV3LCEISjbW9jNLQsuSAV/TznJ3UoEmZh9YMA+cwpBBo
nLcHxRx+0ymq9R53yUCyAUfLWl0pbfUlCF0EKHJ25et/T8TftzMqdLFJ23gyONdj
zgPhc0fNN4fKYqvisg/9V+v0NaVR8VtV1yOhRZp4qy+7JCC6veCb1BZtAVfYJYA4
Mj+k9Una6z6Qnjl97VECAwEAAaBHMEUGCSqGSIb3DQEJDjE4MDYwNAYDVR0RBC0w
K4Ipdm0tMTcxLTEyNC5hYmMuaWRtLmxhYi5lbmcuYnJxLnJlZGhhdC5jb20wDQYJ
KoZIhvcNAQELBQADggEBAImj0y3AaGwCxlkgkGtlou4eookqD/xnaeckmS3cil5M
A6Ff1Ec0orcYBTNBIGwjQN7Z7FC/7ALOa4E94AfqFl6kquy4ArxrVsP3OIC5kGE6
kLcz6F2MJCRuNQdGdxBujlCYyjv4Wr67rLgzeSUD+DGPMehFvdQxTW8VorKBaIxx
tFtqiFZtEs/9bSVfw0jzC4a5maM77kl819I9TXNwwEjGd3/eNmab42qvp9KvXS6Y
hTD+sqE5GmL09YvV3KR1vU1o+9iGJF9Nvg8xOROzTCN+U+TKLv982kbOrrKYwdgn
fYlMwsGlnPuwSNhMH51fxZU5o/F6EXOAdLqIEPEuFQo=
```

Reply:
Sure, happy to. Updated.